crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -7,7 +7,7 @@ use certificate::parse_root_certificate::create_tls_config_without_client_cert;
 use rumqttc::tokio_rustls::rustls::AlertDescription;
 use rumqttc::tokio_rustls::rustls::CertificateError;
 use rumqttc::tokio_rustls::rustls::Error;
-use rumqttc::Client;
+use rumqttc::AsyncClient;
```

Comment:
Rather than using `rumqttc::AsyncClient` it would be better to use `mqtt_channel`. And similarly for the connection checks done through the local broker as well as the JWT token requests. In the later case, even better, we should use the code provided by `c8y_api`.

However, it might be better to do that in a follow up PR, this one being already tedious to review.